pkg/minikube/cluster/cluster.go

```diff
@@ -139,6 +149,20 @@ func StartHost(api libmachine.API, config cfg.MachineConfig) (*host.Host, error)
 	return h, nil
 }
 
+// acquireMachinesLock protects against code that is not parallel-safe (libmachine, cert setup)
+func acquireMachinesLock() (mutex.Releaser, error) {
+	// NOTE: Provisioning generally completes within 60 seconds
+	spec := lock.PathMutexSpec(filepath.Join(localpath.MiniPath(), "machines"))
+	spec.Timeout = 5 * time.Minute
+
+	glog.Infof("acquiring machines lock: %+v", spec)
+	start := time.Now()
+	defer func() {
+		glog.Infof("acquired machines lock within %s", time.Since(start))
```

Comment:
it would be maybe usefull to include the profile name in the log, to find out which profile tried to access the lock.

Reply:
Good suggestion. Done!